### PR TITLE
Fixed all issues psalm, more clean code.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | https://github.com/yiisoft/db/issues/152

Check: `vendor/bin/psalm src/Query`

### CODE REVIEW

File: src/Expression/ExpressionBuilderInterface.php [hunk 1]
```diff
@@ -16,7 +16,7 @@ interface ExpressionBuilderInterface
      * @param ExpressionInterface $expression the expression to be built.
      * @param array $params the binding parameters.
      *
-     * @return string the raw SQL that will not be additionally escaped or quoted.
+     * @return string|array the raw SQL that will not be additionally escaped or quoted.
      */
-    public function build(ExpressionInterface $expression, array &$params = []): string;
+    public function build(ExpressionInterface $expression, array &$params = []);
```
Comment (samdark) [inline src/Expression/ExpressionBuilderInterface.php:21]: Is that possible to make `HashCondition` to always build into string so this type-hint is left as is? It make types more strict in many places.

Comment (terabytesoftw) [reply src/Expression/ExpressionBuilderInterface.php]: Fixed.

File: src/Query/Conditions/BetweenColumnsConditionBuilder.php [hunk 2]
```diff
@@ -41,17 +49,23 @@ public function build(ExpressionInterface $expression, array &$params = []): str
      * @param Query|ExpressionInterface|string $columnName
      * @param array $params the binding parameters.
      *
-     * @return string
+     * @throws Exception|InvalidArgumentException|InvalidConfigException|NotSupportedException
+     *
+     * @return array|string
      */
-    protected function escapeColumnName($columnName, array &$params = []): string
+    protected function escapeColumnName($columnName, array &$params = [])
```
Comment (samdark) [inline src/Query/Conditions/BetweenColumnsConditionBuilder.php:56]: That should always return a string or else code in `build()` will fail.

File: src/Query/Conditions/BetweenColumnsConditionBuilder.php [hunk 3]
```diff
@@ -64,9 +78,11 @@ protected function escapeColumnName($columnName, array &$params = []): string
      * @param mixed $value
      * @param array $params passed by reference
      *
-     * @return string
+     * @throws Exception|InvalidArgumentException|InvalidConfigException|NotSupportedException
+     *
+     * @return array|string
      */
-    protected function createPlaceholder($value, array &$params): string
+    protected function createPlaceholder($value, array &$params)
```
Comment (samdark) [inline src/Query/Conditions/BetweenColumnsConditionBuilder.php:85]: That should always return a string or else code in `build()` will fail.

File: src/Query/Conditions/BetweenConditionBuilder.php [hunk 4]
```diff
@@ -44,9 +52,11 @@ public function build(ExpressionInterface $expression, array &$params = []): str
      * @param mixed $value
      * @param array $params passed by reference
      *
-     * @return string
+     * @throws InvalidArgumentException|Exception|InvalidConfigException|NotSupportedException
+     *
+     * @return array|string
      */
-    protected function createPlaceholder($value, array &$params): string
+    protected function createPlaceholder($value, array &$params)
```
Comment (samdark) [inline src/Query/Conditions/BetweenConditionBuilder.php:59]: That should always return a string or else code in `build()` will fail.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Query/Conditions/BetweenConditionBuilder.php (hunk 4 above)
Comment (terabytesoftw) [reply src/Query/Conditions/BetweenConditionBuilder.php]: fixed.

File: src/Query/Conditions/BetweenColumnsConditionBuilder.php (hunk 3 above)
Comment (terabytesoftw) [reply src/Query/Conditions/BetweenColumnsConditionBuilder.php]: fixed.

File: src/Query/Conditions/BetweenColumnsConditionBuilder.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Query/Conditions/BetweenColumnsConditionBuilder.php]: fixed.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**